lib/datadog/tracing/contrib/propagation/sql_comment.rb

```diff
@@ -0,0 +1,49 @@
+# typed: false
+
+require_relative 'sql_comment/comment'
+require_relative 'sql_comment/ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Propagation
+        # Implements sql comment propagation related contracts.
+        module SqlComment
+          def self.annotate!(span_op, mode)
+            return unless mode.enabled?
+
+            span_op.set_tag(Ext::TAG_DBM_TRACE_INJECTED, true) if mode.full?
+          end
+
+          def self.prepend_comment(sql, span_op, mode, tags: {})
+            return sql unless mode.enabled?
+
+            tags.merge!(service_context)
+            tags.merge!(trace_context(span_op)) if mode.full?
```

Comment:
I recommend commenting out this line for now, so that this can never run until we implement it, even if we mistakenly set it to `full`.